File: bench/data-structures-vector/6.go
package main

import (
	"math/rand"
	"os"
	"sort"
	"strconv"
	"fmt"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())

	var n = 100000000
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.Atoi(args[0])
	}

	var arr = make([]int, n)

	// Random array
	for i := 0; i < n; i++ {
		arr[i] = rand.Int()
	}

	var start = time.Now()

	sort.Ints(arr)

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}

File: bench/data-structures-vector/6.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <ctime>
#include <cstdlib>
#include <chrono>

using namespace std;

int main(int argc, char const *argv[])
{
	srand(time(NULL));

	int S = 100000000;
	if (argc == 2)
	{
		S = atoi(argv[1]);
	}

	vector<int> vec;

	// Random sorted array
	for (int i = 0; i < S; i++)
	{
		vec.push_back(rand());
	}

	auto start = chrono::steady_clock::now();

	sort(vec.begin(), vec.end());

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	return 0;
}
